### PR TITLE
[MWAN/MT/CNI] BGP

### PR DESCRIPTION
**Pull request by marciocloudflare**
### Summary

Closes PCX-11400

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

File: src/content/docs/magic-transit/how-to/advertise-prefixes.mdx [hunk 1]
```diff
@@ -89,6 +88,11 @@ After receiving your information, Cloudflare will update firewall filters to est
 When you withdraw a prefix using BGP, you must ensure the prefix is withdrawn across all BGP sessions on all route reflectors. Otherwise, your prefix will not be withdrawn from Cloudflare's global network.
 :::
 
+### BGP peering
+
+If you use Direct CNI as a way to on-ramp your network traffic to Magic Transit, refer to [BGP peering](/magic-transit/how-to/bgp-peering) to learn how to configure BGP peering.
```
Comment (KianNH) [inline src/content/docs/magic-transit/how-to/advertise-prefixes.mdx]:
```suggestion
If you use Direct CNI as a way to on-ramp your network traffic to Magic Transit, refer to [BGP peering](/magic-transit/how-to/bgp-peering/) to learn how to configure BGP peering.
```

File: src/content/docs/magic-transit/network-interconnect.mdx [hunk 2]
```diff
@@ -22,7 +22,7 @@ When working with Magic Transit and CNI, there are a few guidelines you should f
 
 With [Direct CNI](/network-interconnect/express-cni/) you can use the Cloudflare dashboard to provision a connection to Cloudflare in three minutes or less. This type of connection supports IP packets with 1,500 bytes, both for ingress and egress traffic.
 
-For the time being, Direct CNI does not support BGP. If your use case calls for BGP anycast prefix withdrawal or BGP based connection failover, you need to use [Classic CNI](/network-interconnect/classic-cni/) with a Magic Transit GRE tunnel.
+With Direct CNI you can also setup BGP peering between your network and Cloudflare.
```
Comment (patriciasantaana) [inline src/content/docs/magic-transit/network-interconnect.mdx:25]: Might sound better?
```suggestion
You can also setup BGP peering between your network and Cloudflare with Direct CNI.
```

File: src/content/docs/magic-wan/network-interconnect.mdx [hunk 3]
```diff
@@ -18,7 +18,7 @@ When working with Magic WAN and Cloudflare Network Interconnect (CNI), there are
 
 With [Direct CNI](/network-interconnect/express-cni/) you can use the Cloudflare dashboard to provision a connection to Cloudflare in three minutes or less. This type of connection supports IP packets with 1,500 bytes, both for ingress and egress traffic.
 
-For the time being, Direct CNI does not support BGP. You need to create policy-based routes to ensure traffic is routed correctly. You also need to implement tracking of network traffic to ensure that the CNI and Cloudflare data center is functioning properly.
+With Direct CNI you can also setup BGP peering between your network and Cloudflare.
```
Comment (patriciasantaana) [inline src/content/docs/magic-wan/network-interconnect.mdx:21]: Same as above
```suggestion
You can also setup BGP peering between your network and Cloudflare with Direct CNI.
```

File: src/content/partials/network-interconnect/bgp-peering.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  - legacyHCs
+---
+
+import { Markdown } from "~/components";
+
+{props.productName} customers can use the Cloudflare dashboard to configure and manage BGP peering between their networks and their Magic routing table when using a Direct CNI on-ramp.
+
+Using BGP peering with a CNI allows customers to:
+- Automate the process of adding or removing networks and subnets
+- Take advantage of failure detection and session recovery features
```
Comment (patriciasantaana) [inline src/content/partials/network-interconnect/bgp-peering.mdx]: The bullets under this section have a period after each line, so perhaps to remain consistent? (either or)
```suggestion
- Automate the process of adding or removing networks and subnets.
- Take advantage of failure detection and session recovery features.
```

File: src/content/partials/network-interconnect/bgp-peering.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+## Limitations
+
+BGP multipath is supported. If the same prefix is learned on two different interconnects then traffic destined for that prefix will be distributed across each interconnect according to the usual ECMP behavior.
+
+BGP support currently has the following limitations:
+- The Cloudflare account ASN and the customer device ASN must be different. Only eBGP is supported.
+- Routes are always injected with a priority of 100
+- Bidirectional Forwarding Detection (BFD) is not supported
+- 4-byte ASNs are not supported
```
Comment (patriciasantaana) [inline src/content/partials/network-interconnect/bgp-peering.mdx]:
```suggestion
- The Cloudflare account ASN and the customer device ASN must be different. Only eBGP is supported.
- Routes are always injected with a priority of 100.
- Bidirectional Forwarding Detection (BFD) is not supported.
- 4-byte ASNs are not supported.
```

File: src/content/partials/network-interconnect/bgp-peering.mdx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+{props.productName} customers should configure legacy <a href={props.legacyHCs}>bidirectional health checks</a>.
+
+## How to choose an ASN for BGP peering
+
+The Magic routing table is under the control of the customer, and the customer is able to choose both the Cloudflare side ASN and their customer device side ASN.
+
+By default each BGP peering session will use the same Cloudflare side ASN to represent peering with the Magic WAN/Transit routing table. This default ASN is called the **CF Account ASN** and should be configured to a private 2-byte ASN (64512 and 65534). To set this ASN:
+
+1. Log in to the Cloudflare dashboard, and select your account.
```
Comment (patriciasantaana) [inline src/content/partials/network-interconnect/bgp-peering.mdx]:
```suggestion
1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/), and select your account.
```

File: src/content/partials/network-interconnect/bgp-peering.mdx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+2. Go to Magic WAN / Transit > Configuration > BGP.
```
Comment (patriciasantaana) [inline src/content/partials/network-interconnect/bgp-peering.mdx]:
```suggestion
2. Go to **Magic WAN / Transit** > **Configuration** > **BGP**.
```

File: src/content/partials/network-interconnect/bgp-peering.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+3. In CF Account ASN, enter Cloudflare's ASN.
+
+### For Magic WAN customers
+- The Cloudflare side ASN will be included in the `AS PATH` of announced routes to any BGP enabled interconnect.
+- The customer device ASN can be chosen by the customer, and should be different to the Cloudflare side ASN.
+
+### For Magic Transit customers
+- The Cloudflare side ASN will never be exposed in `AS PATH` of anycast announcements from the Cloudflare edge. In those announcements, Cloudflare will always use the Cloudflare ASN of `13335` optionally prepended with a bring-your-own ASN as described in [Cloudflare ASN vs. your own ASN](/magic-transit/how-to/advertise-prefixes/#cloudflare-asn-vs-your-own-asn)
+- The customer device ASN can be a private ASN, or the ASN they are using for Magic Transit anycast announcements at the edge: this has no impact on the ASN for the anycast announced prefix at the edge of the Cloudflare global network.
```
Comment (patriciasantaana) [inline src/content/partials/network-interconnect/bgp-peering.mdx]:
```suggestion
### For Magic WAN customers

- The Cloudflare side ASN will be included in the `AS PATH` of announced routes to any BGP enabled interconnect.
- The customer device ASN can be chosen by the customer, and should be different to the Cloudflare side ASN.

### For Magic Transit customers

- The Cloudflare side ASN will never be exposed in `AS PATH` of anycast announcements from the Cloudflare edge. In those announcements, Cloudflare will always use the Cloudflare ASN of `13335` optionally prepended with a bring-your-own ASN as described in [Cloudflare ASN vs. your own ASN](/magic-transit/how-to/advertise-prefixes/#cloudflare-asn-vs-your-own-asn)
- The customer device ASN can be a private ASN, or the ASN they are using for Magic Transit anycast announcements at the edge: this has no impact on the ASN for the anycast announced prefix at the edge of the Cloudflare global network.
```

**Review by patriciasantaana — APPROVED** [review]

**Pull request merged**